src/Pages/Details.js

```diff
@@ -55,7 +47,7 @@ const Details = ({ characters }) => {
       )}
       <FavoriteButton
         onClick={toggleFavorites}
-        isFavorite={favoriteIDs.includes(id)}
+        isFavorite={favoriteIDs.includes(currentCharacter.id)}
```

Comment:
Have you tried reloading the Details page in the browser? This line of code should produce a bug, since currentCharacter can be undefined.